ecc/bls12-377/fr/gkr/gkr_test.go

```diff
@@ -728,44 +725,80 @@ func newTestCase(path string) (*TestCase, error) {
 	return tCase, nil
 }
 
-type _select int
+func TestRegisterGateDegreeDetection(t *testing.T) {
+	testGate := func(name string, f func(...fr.Element) fr.Element, nbIn, degree int) {
+		t.Run(name, func(t *testing.T) {
+			name = name + "-register-gate-test"
 
-func (g _select) Evaluate(in ...fr.Element) fr.Element {
-	return in[g]
-}
+			assert.NoError(t, RegisterGate(name, f, nbIn, WithDegree(degree)), "given degree must be accepted")
```

Comment:
Can use unique gate names and then we can omit `RemoveGate`

Reply:
I guess strictly speaking we never needed to remove at all, since `RegisterGate` doesn't prevent writing over an existing gate.